improver/psychrometric_calculations/condensation_trails.py

```diff
@@ -0,0 +1,92 @@
+# (C) Crown Copyright, Met Office. All rights reserved.
+#
+# This file is part of 'IMPROVER' and is released under the BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+"""Module to contain Condensation trail formation calculations."""
+
+from typing import Union
+
+import numpy as np
+from iris.cube import Cube, CubeList
+
+from improver import BasePlugin
+from improver.constants import EARTH_REPSILON
+from improver.utilities.common_input_handle import as_cubelist
+
+
+class CondensationTrailFormation(BasePlugin):
+    """Plugin to calculate whether a condensation trail (contrail) will
+    form based on a given set of atmospheric conditions.
+
+    The calculations require cubes of the following data:
+
+    - Temperature on pressure levels.
+    - Relative Humidity on pressure levels.
+
+    Alongside constants including the ratio of the molecular masses of
+    water and air (EARTH_REPSILON), and defined values for the engine
+    contrail factor.
+
+    References:
+        Schrader, M.L., 1997. Calculations of aircraft contrail
+        formation critical temperatures. Journal of Applied
+        Meteorology, 36(12), pp.1725-1729.
+    """
+
+    def __init__(self, engine_contrail_factors: list = [3e-5, 3.4e-5, 3.9e-5]):
+        """Initialsies the Class"""
+
+        self._engine_contrail_factors: np.ndarray = np.array(
+            engine_contrail_factors, dtype=np.float32
+        )
+
+    def calculate_engine_mixing_ratios(self) -> np.ndarray:
```

Comment:
This probably ought to be a protected method, e.g. 
```suggestion
    def _calculate_engine_mixing_ratios(self) -> np.ndarray:
```